autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectColumn.tsx

```diff
@@ -33,6 +34,44 @@ const CourseSelectColumn: React.FC = () => {
     return (): void => { dispatch(clearCourseCards()); };
   }, [term, dispatch]);
 
+  // When courseCards is changed, save courses (and throttle API calls)
+  React.useEffect(() => {
+    if (!term) return;
+
+    const saveCourses = (): void => {
+      // Serialize courseCards and make API call
+      const courses: SerializedCourseCardOptions[] = [];
+      for (let i = 0; i < courseCards.numCardsCreated; i++) {
+        const course = courseCards[i];
+        if (course) {
+          const sections = course.sections.filter(({ selected }) => selected).map((sectionSel) => (
+            sectionSel.section.id
+          ));
+          courses.push({
+            course: course.course,
+            customizationLevel: course.customizationLevel,
+            honors: course.honors,
+            web: course.web,
+            sections,
+          });
+        }
+      }
+
+      const body = JSON.stringify({
+        courses,
+        term,
+      });
+
+      fetch('sessions/save_courses', {
+        method: 'PUT',
+        body,
+      });
```

Comment:
```suggestion
      fetch('sessions/save_courses', {
        method: 'PUT',
        JSON.stringify({ courses, term, }),
      });
```

I think I'd prefer if you shortened it to this form, but really just a nitpick so feel free to disregard